### PR TITLE
DDF-1164 Anonymous interceptor does not work correctly

### PR DESCRIPTION
**Pull request by stustison**
...y and set the soap action, fixed the wrapper to pick up the interceptor, fixed the security assertion store to work even if the token hasn't been cached yet

### CODE REVIEW

File: security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java [hunk 1]
```diff
@@ -37,14 +42,26 @@ public AnonymousInterceptorWrapper() {
...
+            return cxfBundle.getBundleContext();
+        }
+        return null;
     }
 
     @Override
 	public void handleMessage(SoapMessage msg) throws Fault {
+        BundleContext context = getContext();
+        PhaseInterceptor anonIntercep = null;
+        ServiceReference anonIntercepRef = context.getServiceReference(PhaseInterceptor.class.getName());
```
Comment (kcwire) [inline security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java]: Are we sure we only want the highest ranked PhaseInterceptor? or should we add some filter such that we are sure it's the AnonymousInterceptor.

Comment (pklinef) [inline security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java]: Is there a reason we cannot do a service reference list for this?  Then we do not need the bundle context.

File: security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java [hunk 2]
```diff
@@ -37,14 +42,26 @@ public AnonymousInterceptorWrapper() {
...
+
+        if (anonIntercepRef != null) {
+            anonIntercep = (PhaseInterceptor) context.getService(anonIntercepRef);
```
Comment (kcwire) [inline security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java]: Could be null or throw some exception. Should probably protect against that.
https://osgi.org/javadoc/r4v43/core/org/osgi/framework/BundleContext.html#getService(org.osgi.framework.ServiceReference)

File: security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java (hunk 1 above)
Comment (jvettraino) [inline security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java]: I would definitely say you want a filter.  This came up as a potential area of concern last time I was looking at this.

File: security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java [hunk 3]
```diff
@@ -293,8 +297,21 @@ public void handleMessage(SoapMessage message) throws Fault {
...
+                                    String xpath = "/SignedSupportingTokens/Policy/ExactlyOne/All/IssuedToken/RequestSecurityTokenTemplate/TokenType";
                                     tokenType = retrieveXmlValue(xml, xpath);
+                                    if (StringUtils.isEmpty(tokenType)) {
+                                        xpath = "/SignedSupportingTokens/Policy/IssuedToken/RequestSecurityTokenTemplate/TokenType";
+                                        tokenType = retrieveXmlValue(xml, xpath);
+                                    }
+                                    supportingTokenAssertion = qName;
+
+                                } else if (qName.equals(SP12Constants.SUPPORTING_TOKENS)) {
+                                    String xpath = "/SupportingTokens/Policy/ExactlyOne/All//IssuedToken/RequestSecurityTokenTemplate/TokenType";
```
Comment (kcwire) [inline security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java]: Double "/" after "All//"?

Looks like it should be the same as line 300 - maybe define these as constants?

File: security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java [hunk 4]
```diff
@@ -293,8 +297,21 @@ public void handleMessage(SoapMessage message) throws Fault {
                                     requireClientCert = evaluateExpression(xml, xpath);
 
                                 } else if (qName.equals(SP12Constants.SIGNED_SUPPORTING_TOKENS)) {
-                                    String xpath = "/SignedSupportingTokens/Policy/IssuedToken/RequestSecurityTokenTemplate/TokenType";
+                                    String xpath = "/SignedSupportingTokens/Policy/ExactlyOne/All/IssuedToken/RequestSecurityTokenTemplate/TokenType";
```
Comment (pklinef) [inline security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java]: You should be able to modify this xpath with a wild card to work for both cases.

**Comment by pklinef** [comment]
:+1:

**Comment by kcwire** [comment]
+1 after changes

**Pull request merged**